src/rgw/rgw_swift_auth.cc

```diff
@@ -524,8 +524,10 @@ void RGW_SWIFT_Auth_Get::execute()
       protocol = "https";
     } else {
       server_port = s->info.env->get("SERVER_PORT");
-      add_port = (strcmp(server_port, "80") != 0);
-      protocol = "http";
+      if (server_port) {
```

Comment:
the url should always start with a protocol, either "https" or "http" here.

@yehudasa shall we have a default 80 "SERVER_PORT"  here, so we can default to "http" here?